src/components/views/globals/MatrixToolbar.js

```diff
@@ -32,6 +37,7 @@ module.exports = React.createClass({
     },
 
     render: function() {
+        if (this.state.showToolbar != true) return null;
```

Comment:
I don't think it's legal to return `null` here. But anyway, this logic is in the wrong place; the logic for whether to show the toolbar or not belongs in the parent component (MatrixChat).

`MatrixChat.render` already has: 
```
... if (Notifier.supportsDesktopNotifications() && !Notifier.isEnabled() && !Notifier.isToolbarHidden()) {
            topBar = <MatrixToolbar />;
}
```

You just need to update `Notifier.isToolbarHidden`.